[17.0][MIG] web_widget_remote_measure: Migration to 17.0

chienandalu opened the pull request:
- [x] Refactor js field widget to owl field component.
- [x] Test tcp connection


cc @Tecnativa TT52446

cc  @CarlosRoca13 @sergio-teruel 

Gists for testing locally

- websockets: https://gist.github.com/chienandalu/9d207b977f5681c036a7d3bda27109b8
- tcp: https://gist.github.com/chienandalu/b1eb4a93aa743b2e165eab22f2e75cdc

chienandalu committed:
[ADD] web_widget_remote_measure: New module

TT38745

chienandalu committed:
[IMP] web_widget_remote_measure: add tcp direct connection

TT47128

chienandalu committed:
[IMP] web_widget_remote_measure: non stop read

TT49230

chienandalu committed:
[MOV] web_widget_remote_measure: move from OCA/web

Commit:
[UPD] Update web_widget_remote_measure.pot

chienandalu committed:
[FIX] web_widget_remote_measure: avoid fail on no device

Error on property when the widget doesn't have a device data

chienandalu committed:
[IMP] web_widget_remote_measure: pre-commit auto fixes

chienandalu committed:
[MIG] web_widget_remote_measure: Migration to 17.0

TT52446

CarlosRoca13 reviewed (approved):
Code review :smile: Thanks!

pedrobaeza commented:
/ocabot migration web_widget_remote_measure

sergio-teruel committed:
[IMP] web_widget_remote_measure: Add button to test connection with the device
TT55802

CarlosRoca13 commented:
Added changes introduced on https://github.com/OCA/stock-weighing/pull/34

CarlosRoca13 committed:
[IMP] web_widget_remote_measure: Allow quick selection of the user's remote device

Sometimes, a user may be working from different devices (if they're an operator) or moving around the company with a tablet and needs to weigh using the nearest device, without losing focus on their current task.

With these changes, we allow them to quickly switch remote devices without having to access their user profile.

pedrobaeza reviewed (approved):
/ocabot merge nobump
